projects/plugins/jetpack/extensions/blocks/donations/tab.js

```diff
@@ -1,166 +0,0 @@
-/**
- * WordPress dependencies
- */
-import { RichText } from '@wordpress/block-editor';
-import { useSelect } from '@wordpress/data';
-import { __, sprintf } from '@wordpress/i18n';
-import { useEffect, useState, useMemo } from '@wordpress/element';
-
-/**
- * Internal dependencies
- */
-import Amount from './amount';
-import { minimumTransactionAmountForCurrency } from '../../shared/currencies';
-
-const Tab = ( { activeTab, attributes, setAttributes } ) => {
-	const {
-		currency,
-		oneTimeDonation,
-		monthlyDonation,
-		annualDonation,
-		showCustomAmount,
-		chooseAmountText,
-		customAmountText,
-	} = attributes;
-
-	const donationAttributes = {
-		'one-time': 'oneTimeDonation',
-		'1 month': 'monthlyDonation',
-		'1 year': 'annualDonation',
-	};
-
-	const getDonationValue = key => attributes[ donationAttributes[ activeTab ] ][ key ];
-
-	const setDonationValue = ( key, value ) => {
-		const donationAttribute = donationAttributes[ activeTab ];
-		const donation = attributes[ donationAttribute ];
-		setAttributes( {
-			[ donationAttribute ]: {
-				...donation,
-				[ key ]: value,
-			},
-		} );
-	};
-
-	// Updates the amounts whenever there are new defaults due to a currency change.
-	const [ previousCurrency, setPreviousCurrency ] = useState( currency );
-	const minAmount = minimumTransactionAmountForCurrency( currency );
-	const defaultAmounts = useMemo(
```

Comment:
Very minor - maybe a follow-up issue if at all, but some behaviour has changed and I think it's related to this code.

Previously if I inserted a donations block it'd have a default of e.g. $5, when I changed the currency to JPY it changed it to 500 yen, see https://d.pr/v/YXV18n. On this branch it doesn't change the amount, just the symbol, so it is 5 yen.

Reply:
When the currency changes we could do a hard reset of whatever amounts the user has inputted, that would be pretty easy and is how the previous version behaves as can be seen here:

https://user-images.githubusercontent.com/1989914/153249335-99ef602b-0e35-4839-a01d-1fdfe821f0bf.mp4

But I'm not sure if that is an ideal UX, for example, I consider a typical case to be that I start out using the donations block and I change the fields to let's say: 10, 25, 50. Then I realize that the currency is in USD (default currency), I would then need to change the currency to Euros and all my fields would get reset. 
With the current approach, I would maintain the values that I've inputted instead of starting fresh again.
I don't have a strong opinion on which approach is best though.

Reply:
After exploring the issue some more, I've made the current implementation work as the previous implementation. With the current implementation, if we change from USD to JPY, the amount fields are in an inconsistent state that can be persisted. That is, we can save amounts below the minimum donation amount for the given currency, which would then cause the donation button to not work as expected.